From a98d3ada2a4d51bd5fbd676fe3306871ad8228eb Mon Sep 17 00:00:00 2001
From: Carlos Goncalves <carlos.goncalves@neclab.eu>
Date: Thu, 20 Apr 2017 17:52:43 +0000
Subject: [PATCH] Add data plane status support to Network Port obj

Added 'data_plane_status' parameter to Port class.

Partial-Bug: #1684989
Change-Id: I716ee25d1e7e4f81319f66b7f7457db243b4ffe3
---
 openstack/network/v2/port.py                 | 2 ++
 openstack/tests/unit/network/v2/test_port.py | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/openstack/network/v2/port.py b/openstack/network/v2/port.py
index 6234ea45..e98de374 100644
--- a/openstack/network/v2/port.py
+++ b/openstack/network/v2/port.py
@@ -67,6 +67,8 @@ class Port(resource.Resource):
     binding_vnic_type = resource.Body('binding:vnic_type')
     #: Timestamp when the port was created.
     created_at = resource.Body('created_at')
+    #: Underlying data plane status of this port.
+    data_plane_status = resource.Body('data_plane_status')
     #: The port description.
     description = resource.Body('description')
     #: Device ID of this port.
diff --git a/openstack/tests/unit/network/v2/test_port.py b/openstack/tests/unit/network/v2/test_port.py
index b61418ce..dd931a76 100644
--- a/openstack/tests/unit/network/v2/test_port.py
+++ b/openstack/tests/unit/network/v2/test_port.py
@@ -24,6 +24,7 @@ EXAMPLE = {
     'binding:vif_type': '6',
     'binding:vnic_type': '7',
     'created_at': '2016-03-09T12:14:57.233772',
+    'data_plane_status': '32',
     'description': '8',
     'device_id': '9',
     'device_owner': '10',
@@ -82,6 +83,7 @@ class TestPort(testtools.TestCase):
         self.assertEqual(EXAMPLE['binding:vif_type'], sot.binding_vif_type)
         self.assertEqual(EXAMPLE['binding:vnic_type'], sot.binding_vnic_type)
         self.assertEqual(EXAMPLE['created_at'], sot.created_at)
+        self.assertEqual(EXAMPLE['data_plane_status'], sot.data_plane_status)
         self.assertEqual(EXAMPLE['description'], sot.description)
         self.assertEqual(EXAMPLE['device_id'], sot.device_id)
         self.assertEqual(EXAMPLE['device_owner'], sot.device_owner)
-- 
2.12.3

